src/services/balancer/subgraph/entities/pools/index.ts

```diff
@@ -56,10 +63,41 @@ export default class Pools {
       pool.totalLiquidity = getPoolLiquidity(pool, prices);
       const pastPool = pastPools.find(p => p.id === pool.id);
       const volume = this.calcVolume(pool, pastPool);
-      const apy = this.calcAPY(pool, pastPool);
+      const poolAPY = this.calcAPY(pool, pastPool);
       const fees = this.calcFees(pool, pastPool);
 
-      return { ...pool, dynamic: { period, volume, apy, fees } };
+      let liquidityMiningAPY = '0';
+      const liquidityMiningRewards = currentLiquidityMiningRewards[pool.id];
+      const hasLiquidityMiningRewards = IS_LIQUIDITY_MINING_ENABLED
+        ? !!liquidityMiningRewards
+        : false;
+
+      if (hasLiquidityMiningRewards) {
+        liquidityMiningAPY = computeAPYForPool(
+          liquidityMiningRewards,
+          prices[TOKENS.AddressMap.BAL].price || 0,
+          pool.totalLiquidity
+        );
+      }
```

Comment:
I'd move this into it's own function like `calcAPY` to help keep this serialize function as simple as possible to understand.